### PR TITLE
Implemented compute_average_structure()

### LINKED ISSUE #1159: Consensus structure
It would be nice if mdtraj could be given an ensemble of similar structures and find a "consensus" structure that is an average of them. I saw this paper http://www.ncbi.nlm.nih.gov/pubmed/17369627

I'm not sure about the current literature for this and/or whether this functionality is available in other packages

### PR DESCRIPTION
**Pull request by peastman**
Fixes #1159 and #1218.

### CODE REVIEW

**Comment by peastman** [comment]
Does this look ok to everyone?  Shall I merge it?

File: mdtraj/geometry/alignment.py [hunk 1]
```diff
@@ -269,3 +269,33 @@ def rmsd_qcp(conformation1, conformation2):
     max_eigenvalue = scipy.optimize.newton(f, E0, df)
     rmsd = np.sqrt(np.abs(2.0 * (E0 - max_eigenvalue) / n_atoms))
     return rmsd
+
+def compute_average_structure(xyz):
+    """Compute the average structure from a set of frames.
+
+    The frames are first aligned to minimize the RMSD from the average structure,
```
Comment (jchodera) [inline mdtraj/geometry/alignment.py]: Why not add an optional argument to allow users to specify whether or not the frames should first be RMSD-aligned? I could imagine both schemes could be useful.

File: mdtraj/geometry/alignment.py [hunk 2]
```diff
@@ -269,3 +269,33 @@ def rmsd_qcp(conformation1, conformation2):
...
+    n_atoms = xyz.shape[1]
+    candidate = xyz[0]
+    
+    # In practice there is usually negligible improvement after the second iteration.
+    # We do three just to be safe.
+    
+    for iteration in range(3):
+        average = np.zeros((n_atoms, 3))
+        for frame in range(n_frames):
+            average += transform(xyz[frame], candidate)
```
Comment (jchodera) [inline mdtraj/geometry/alignment.py]: Are you just aligning to the first frame? Is this approach equivalent to selecting the reference structure for alignment that minimizes the RMSD, as the docstring suggests?

Comment (peastman) [inline mdtraj/geometry/alignment.py]: See #1159.  We discussed this at some length there.

Comment (jchodera) [inline mdtraj/geometry/alignment.py]: Instead of hardcoding three iterations (is there a proof for that?) why not check for actual convergence?

**Comment by mpharrigan** [comment]
I'm going to merge this, @jchodera feel free to submit an additional pr to address the small issues you picked out

**Pull request merged**